server/game/core/ability/abilityTargets/PlayerTargetResolver.ts

```diff
@@ -0,0 +1,83 @@
+import { TargetResolverBaseClass } from './TargetResolverBaseClass';
+import { IPlayerTargetResolver } from '../../../TargetInterfaces';
+import { AbilityContext } from '../AbilityContext';
+import type Player from '../../Player';
+import PlayerOrCardAbility from '../PlayerOrCardAbility';
+import { Stage } from '../../Constants';
+
+export class PlayerTargetResolver extends TargetResolverBaseClass<IPlayerTargetResolver<AbilityContext>> {
+    public constructor(name: string, properties: IPlayerTargetResolver<AbilityContext>, ability: PlayerOrCardAbility) {
+        super(name, properties, ability);
+
+        if (this.properties.immediateEffect) {
+            this.properties.immediateEffect.setDefaultTargetFn((context) => context.targets[name]);
+        }
+    }
+
+    protected override hasLegalTarget(context: any): boolean {
+        // I don't think there are cases of player-targeting abilities that are sometimes invalid, but I could be wrong.
```

Comment:
We need to check here because it depends on what gamesystems are involved. E.g., if there was some card that said "chosen player cannot draw cards for this phase" and this targetResolver was wrapping the "DrawCardSystem", this check could fail (assuming that player is the only legal target due to some filter). Same for `getAllLegalTargets`.

One thing to consider would be making a stripped-down version of `BaseCardSelector` that manages player targets. That would also be useful because we could eventually extend it with additional options like "canChooseMultiple" so that maybe Yoda would have checkboxes instead of an awkward set of buttons like it does now.

That could be a TODO though, but definitely we need some checks here either way.

Reply:
I put a big TODO at the top of the class to create a PlayerSelector that can determine target legality, and marked a couple functions in the class as placeholder for now.